base/iostream.jl

```diff
@@ -213,23 +213,23 @@ fdio(fd::Integer, own::Bool=false) = fdio(string("<fd ",fd,">"), fd, own)
 
 
 """
-    open(filename::AbstractString, [read::Bool, write::Bool, create::Bool, truncate::Bool, append::Bool]) -> IOStream
+    open(filename::AbstractString; write::Bool = true, read::Bool = !write, create::Bool = true, truncate::Bool = true, append::Bool = true) -> IOStream
 
 Open a file in a mode specified by five boolean arguments. The default is to open files for
 reading only. Return a stream for accessing the file.
 """
-function open(fname::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool)
+function open(fname::AbstractString; write::Bool = false, read::Bool = !write, create::Bool = false, truncate::Bool = false, append::Bool = false)
```

Comment:
There should be some relationship between `truncate` and `append`. Probably `truncate = !append`. Otherwise the behavior of `open(file, append=true)` is going to be rather surprising. I also suspect that `write = append` makes sense. Similarly, `create` typically default to true if `write` is set. I think you'll need to draw a graph of the relationships here to figure this one out – this is why this change hasn't been done yet.

Reply:
I went ahead and did it, although this probably could have been done without the image:
![open_keywords](https://user-images.githubusercontent.com/153596/35220511-9f5de5ea-ff44-11e7-9a4c-b1ea5fceba8e.jpg)
So it looks like the signature should be something like this:
```jl
open(fname::AbstractString;
    append::Bool = false,
    write::Bool = append,
    read::Bool = !write,
    create::Bool = write,
    truncate::Bool = !append,
)
```

Reply:
Actually, that's not quite right since you want `truncate = write  && !append`.

Reply:
ok so 
```julia
open(fname::AbstractString;
    append::Bool = false,
    write::Bool = append,
    read::Bool = !write,
    create::Bool = write,
    truncate::Bool = write && !append,
)
```
?